### PR TITLE
[editor][server] 1/n endpoint clear outputs

### PR DESCRIPTION
**Pull request by Ankush-lastmile**
[editor][server] 1/n endpoint clear outputs

server endpoint to clear outputs from server side aiconfig state (changes not persisted to disk).

## Testplan

1. Run Prompt
Screenshot 2024-01-08 at 10 43 49 AM

2. Clear outputs.

Screenshot 2024-01-08 at 10 54 12 AM

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/lastmile-ai/aiconfig/pull/790).
* #791
* __->__ #790

### CODE REVIEW

File: python/src/aiconfig/editor/server/server.py [hunk 1]
```diff
@@ -457,3 +457,30 @@ def set_description() -> FlaskResponse:
...
+        return HttpResponseWithAIConfig(
+            message="No AIConfig in memory, can't run clear outputs.",
+            code=400,
+            aiconfig=None,
+        ).to_flask_format()
+
+    for prompt in aiconfig.prompts:
+        prompt_name = prompt.name
+        # fn name `delete_output`` is misleading. TODO: Rename to `delete_outputs`` in AIConfig API
+        aiconfig.delete_output(prompt_name)
```
Comment (rossdanlm) [inline python/src/aiconfig/editor/server/server.py:480]: nit: would prefer this be part of the SDK itself, but we can make a bootcamp issue for this, not a blocker

File: python/src/aiconfig/editor/server/server.py [hunk 2]
```diff
@@ -457,3 +457,30 @@ def set_description() -> FlaskResponse:
     operation = make_op_run_method(MethodName("set_description"))
     operation_args: Result[OpArgs, str] = result.Ok(OpArgs({"description": description}))
     return run_aiconfig_operation_with_op_args(aiconfig, "set_description", operation, operation_args)
+
+@app.route("/api/clear_outputs", methods=["POST"])
```
Comment (rossdanlm) [inline python/src/aiconfig/editor/server/server.py:461]: To extend, we can take in list of prompt_names, and if none specified, default to all of them (like what we have below)

File: python/src/aiconfig/editor/server/server.py [hunk 3]
```diff
@@ -457,3 +457,30 @@ def set_description() -> FlaskResponse:
...
+            aiconfig=None,
+        ).to_flask_format()
+
+    for prompt in aiconfig.prompts:
+        prompt_name = prompt.name
+        # fn name `delete_output`` is misleading. TODO: Rename to `delete_outputs`` in AIConfig API
+        aiconfig.delete_output(prompt_name)
+
+
+    return HttpResponseWithAIConfig(
```
Comment (rossdanlm) [inline python/src/aiconfig/editor/server/server.py:483]: Jonathan already built `run_aiconfig_operation_with_op_args()`, let's use that

**Review by rossdanlm — APPROVED** [review]
left a few comments

File: python/src/aiconfig/editor/server/server.py (hunk 2 above)
Comment (rholinshead) [reply python/src/aiconfig/editor/server/server.py:461]: That would be handy for clearing single outputs later, but can be done at that point, I think

File: python/src/aiconfig/editor/server/server.py (hunk 3 above)
Comment (jonathanlastmileai) [reply python/src/aiconfig/editor/server/server.py:483]: I think run_aiconfig_operation_with_op_args() is not quite general enough.

 This is fine, but it turns out this should be easy enough like this:
 
 ```
 def _op(aiconfig_runtime: AIConfigRuntime, _op_args: OpArgs) -> Result[None, str]:
        for prompt in aiconfig_runtime.prompts:
            prompt_name = prompt.name
            # fn name `delete_output`` is misleading. TODO: Rename to `delete_outputs`` in AIConfig API
            aiconfig_runtime.delete_output(prompt_name)
        return Ok(None)
    signature: dict[str, Type[Any]] = {}
    return run_aiconfig_operation_with_request_json(aiconfig, request_json, f"method_{method_name}", _op, signature)
 ```

**Review by jonathanlastmileai — APPROVED** [review]

File: python/src/aiconfig/editor/server/server.py (hunk 3 above)
Comment (Ankush-lastmile) [reply python/src/aiconfig/editor/server/server.py]: updated. Thanks @jonathanlastmileai for the help, used `run_aiconfig_operation_with_request_json`.

**Comment by Ankush-lastmile** [comment]
updated to use `run_aiconfig_operation_with_request_json`. Tested with similar testplan as before, request now required to have a valid body. Also tested with frontend on diff on top

**Pull request merged**